pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -1795,4 +1795,10 @@ type ScrapeClass struct {
 	// TLSConfig section for scrapes.
 	// +optional
 	TLSConfig *TLSConfig `json:"tlsConfig,omitempty"`
+
+	// ExtraRelabelings allow setting extra default relabelings to all
+	// scrape objects.
+	//
+	// +optional
+	ExtraRelabelings []*RelabelConfig `json:"extraRelabelings,omitempty"`
```

Comment:
I'm not sure why we went with []*RelabelConfig in the first place for service/pod monitors but I'd rather use []RelabelConfig.

Reply:
The issue with this is that the generator function requires an array of pointers:
`func generateRelabelConfig(rc []*monitoringv1.RelabelConfig) []yaml.MapSlice`

Reply:
Right, let's keep it like that for now and tackle my remark in another PR.